src/js/jsx/shared/Datalist.jsx

```diff
@@ -451,6 +475,15 @@ define(function (require, exports, module) {
             }
         },
 
+        removeAutofillSuggestion: function () {
+            if (this.props.useAutofill) {
+                this.setState({
+                    suggestTitle: ""
+                });
+                this.forceUpdate();
```

Comment:
Otherwise in `shouldComponentUpdate`, it sets `suggestTitle` to a value other than `""` again before rendering.